src/codeu/chat/client/core/Controller.java

```diff
@@ -239,4 +239,53 @@ public void setConversationExplicitPermissions(Uuid convoID, Uuid invoker, Uuid
       LOG.error(ex, "Exception during call on server.");
     }
   }
+  
+  @Override
+  public void deleteMessage(Uuid convoId, Uuid messageId) throws SecurityViolationException {
+	  try (final Connection connection = source.connect()) {
+		  Serializers.INTEGER.write(connection.out(), NetworkCode.DELETE_MESSAGE_REQUEST);
+		  Uuid.SERIALIZER.write(connection.out(), convoId);
+	      Uuid.SERIALIZER.write(connection.out(), messageId);
+	      Uuid.SERIALIZER.write(connection.out(), user.id);
+	      
+	      LOG.info("DeleteMessage: Request completed.");
+	      int returnCode = Serializers.INTEGER.read(connection.in());
+	      if (returnCode == NetworkCode.DELETE_MESSAGE_RESPONSE) {
+	        LOG.info("DeleteMessage: Response completed.");
+	      } else if (returnCode == NetworkCode.ERR_SECURITY_VIOLATION) {
+	        throw new SecurityViolationException();
+	      } else {
+	        LOG.error("Response from server failed.");
+	      }
+	  } catch (SecurityViolationException e) {
+	    throw e;
+	  } catch (Exception ex) {
```

Comment:
We were following the (anti-)pattern of the original code, of logging the exceptions that could possibly be thrown during I/O (broken pipe, connection reset, etc.) with a blanket `Exception`. As you say, I will change it to an `IOException`.